include/yaramod/types/expressions.h

```diff
@@ -1169,6 +1169,70 @@ class OfExpression : public ForExpression
 	virtual TokenIt getLastTokenIt() const override { return _iterable->getLastTokenIt(); }
 };
 
+/**
+ * Class representing an iterator, which is an array of expressions typically
+ * used with an of operator
+ *
+ * For example:
+ * @code
+ * all of [ true, false, true ]
+ *        ^^^^^^^^^^^^^^^^^^^^^
+ * @endcode
+ */
+class IteratorExpression : public Expression
```

Comment:
Is _iterator_ the correct name to choose for this abstraction? _Iterator_ is usually something which iterates over _iterable_. The difference here is that iterator carries notion of what is current element and which one is the next one. This sequence doesn't carry anything like this so is it really `Iterator` or rather `Iterable`?